typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -58,21 +65,24 @@ async function main() {
       };
       try {
         await sendMessage(app, source, destination);
-        messagesCount.labels({ ...labels, status: 'success' }).inc();
+        messagesSendStatus.labels({ ...labels }).set(1);
```

Comment:
Will we be able to set a useful alert based on this gauge? It seems that a failed message, which sets the gauge to zero, can be very quickly followed up by a successful message, which sets the gauge back to 1

Reply:
If there was a successful one do we care that the other failed? We really only care that it is succeeding. Also it only attempts to send one per cron interval which is generally quite high.

Reply:
> Yes, we care if any of them fail, 100% of Kathy messages should get delivered and we need to know if that's ever not the case.

well unless we decide to retry messages automatically we will still be able to alert on any failure as there is a large window where it will be reported as `0`.

> I'm not sure I follow, are you saying that Kathy only sends one message per cronjob? IIUC that's not true at the moment, Kathy appears to be configurable to send many messages.

We only send one message per cron job per origin/remote pairing. There are multiple messages sent but only one per that unique pairing, and the metric cardinality is defined by the pairing so that each origin/remote gets its own gauge value.